Sources/SwiftFormat/Core/RuleMask.swift

```diff
@@ -95,6 +95,34 @@ extension SourceRange {
   }
 }
 
+/// Represents the kind of ignore directive encountered in the source.
+enum IgnoreDirective: CustomStringConvertible {
+  /// A node-level directive that disables rules for the following node and its children.
+  case node
+  /// A file-level directive that disables rules for the entire file.
+  case file
+
+  var description: String {
+    switch self {
+    case .node:
+      return "swift-format-ignore"
+    case .file:
+      return "swift-format-ignore-file"
+    }
+  }
+
+  /// Regex pattern to match an ignore comment. This pattern supports 0 or more comma delimited rule
+  /// names. The rule name(s), when present, are in capture group #3.
+  private var pattern: String {
+    return #"^\s*\/\/\s*"# + description + #"((:\s+(([A-z0-9]+[,\s]*)+))?$|\s+$)"#
+  }
+
+  /// Rule ignore regex object.
+  fileprivate var regex: NSRegularExpression {
+    return try! NSRegularExpression(pattern: pattern, options: [])
+  }
```

Comment:
Since you are working on this already, I’ve got a few thoughts:
- The regex restriction after the `:` seems a little restrictive to me. Eg. I imagine it would be fine to try and debug why a future rule that contains a `_` cannot be ignored using a directive. Maybe we can just make the regex a prefix match. @allevato Do you remember why the regex is this specific?
- This re-compiles the regex every time `regex` is accessed. Compiling regular expressions is fairly expensive, so I’d prefer to cache that.
- Could we use Swift’s Regex instead of `NSRegularExpression`?
- Or, alternatively, would it be easier to write a simple line matcher and avoid going through regex matching at all?

All except for avoiding to re-compile the regex don’t need to be in this PR.

Reply:
Yes, I have updated the implementation to cache the regex instead of recompiling it each time.

And as for other suggestions, would it be a good idea to create a separate issue for these?
- Relaxing the regex restrictions
(If we can raise the minimum deployment version, migrating from `NSRegularExpression` to Swift’s `Regex`)
- Alternatively, parsing ignore directives using string matching instead of regex

Reply:
Issues (or PRs 😉) for those items would be good, yes. Also, I am happy to raise the deployment target to macOS 13. SourceKit-LSP, for example, already requires macOS 13.0.

Reply:
Got it! I'll work on a follow-up PR for those items 😁